bin/install

```diff
@@ -262,6 +266,121 @@ install_g() {
   chmod +x "$GOPATH/bin/g"
 }
 
+#
+# Check whether 'g' alias is already used, setup alternatives alias if it is
+#
+
+configure_g_alias() {
+  if ! g_alias_defined "$G_ALIAS"; then
+    return 0
+  fi
+
+  G_ALIAS_CHANGED=true
+
+  if [ "$NON_INTERACTIVE" = true ]; then
+    G_ALIAS="ggovm"
+    return 0
+  fi
+
+  while g_alias_defined "$G_ALIAS"; do
+    if ! prompt_set_alias "$G_ALIAS" "$G_ALIAS_USED_IN_SHELL"; then
+      echo
+      echo
+      log_info "info" "skipping setup alias for g"
+      G_ALIAS_CHANGED=false
+      break
+    fi
+
+    G_ALIAS=$(set_alias_name)
+  done
+}
+
+#
+# Iteratively check every selected shells for defined alias
+#
+
+g_alias_defined() {
+  alias_already_used=false
+
+  for shell in $SELECTED_SHELLS; do
+    if exists $shell "$1"; then
+      alias_already_used=true
+      G_ALIAS_USED_IN_SHELL="$shell"
+    fi
+  done
+
+  if [ "$alias_already_used" = true ]; then
+    return 0
+  fi
+
+  return 1
+}
+
+#
+# Check whether command already exists
+#
+
+exists() {
+  case "$1" in
```

Comment:
Should we skip the alias configuration for all shells or only those shells are not supported?

Reply:
Not really following.

So the deal here is that, as it stands now, if you try to install g in a shell diff from the ones here (zsh, bash, fish), for example, in ash, you will be promoted to setup an alias in ash, even tho we don't support it (yet, hopefully). And you will also be stuck in a loop of prompts.

This because exit code 0 means "does exist" while exit code 1 means "does **not** exist".

Since there's no defuault case, when the shell is not supported the default exit code is 0.

So an `*) return 1;;` at the end would help fix these cases.

Reply:
Maybe we should invert the exit codes and add explicit returns inside if statements to make it clearer for future readers. 🤔

Reply:
Or am I missing something? I'm reviewing from my phone so I don't have the big picture. 😄

Reply:
Oh, I get it. I initially thought that the alias won't be set for all shells if there's an unsupported shell. So, basically `return 1` as default so the installer won't recognize it as already exists alias, right?  :thinking:

Reply:
That. I think we are on the same page now. :+1:

Reply:
Allrite, I'm already on it